crates/ruma-client-api/src/r0/session/login/fallback.rs

```diff
@@ -0,0 +1,52 @@
+//! [/_matrix/static/client/login/](https://spec.matrix.org/unstable/client-server-api/#login-fallback)
+
+use ruma_api::ruma_api;
+use ruma_identifiers::DeviceId;
+
+ruma_api! {
+    metadata: {
+        description: "Get login fallback web page.",
+        method: GET,
+        name: "login_fallback",
+        path: "/_matrix/static/client/login/",
+        rate_limited: false,
+        authentication: None,
+    }
+
+    request: {
+        /// ID of the client device
+        #[ruma_api(query)]
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub device_id: Option<&'a DeviceId>,
+
+        /// A display name to assign to the newly-created device. Ignored if device_id corresponds
+        /// to a known device.
+        #[ruma_api(query)]
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub initial_device_display_name: Option<&'a str>,
+    }
+
+    #[derive(Default)]
```

Comment:
I don't think there should be a `Default` implementation here. Why would one want to return an empty page from this endpoint?

Reply:
hmmm, good point, i'll remove that